### PR TITLE
Improve performance of "snabb config get" on big configurations

### PR DESCRIPTION
**Pull request by wingo**
This change also adds a --trace option that can log incoming RPCs to a ptree manager.  It's off by default as it does impose a bit of overhead.

### CODE REVIEW

File: src/lib/ptree/trace.lua [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+end
+
+function selftest ()
+   print('selftest: lib.ptree.trace')
+   local S = require('syscall')
+
+   local tmp = os.tmpname()
+   local trace = new({file=tmp})
+   trace:record("foo", {bar="baz"})
+   trace:record("qux", {bar="baz", zog="100"})
```
Comment (dpino) [inline src/lib/ptree/trace.lua]: The test fails cause `trace:record` calls `listen_directive_for_rpc` which only admits a limited number of verbs and args. Is either necessary to change the test arguments or broad `listen_directive_for_rpc` to admit more verbs and args, dunno.

File: src/lib/yang/util.lua [hunk 2]
```diff
@@ -87,12 +87,47 @@ function ipv4_ntop(addr)
...
-   function file:write(str) table.insert(out, str) end
-   function file:flush(str) return table.concat(out) end
-   function file:clear(str) out = {} end
+   local size = 1024
+   local buf = alloc(size)
+   local written = 0
+   local read = 0
+   function file:write(str)
+      while size - written < #str do
+         if 0 < read then
```
Comment (dpino) [inline src/lib/yang/util.lua:107]: I have a doubt here. I think the `else` branch is always taken. The `then` branch is taken when read is greater than 0, but `read` is initially as zero and when it gets updated (in `write` and `reset`) is set to zero. So what actually happens in the end is that the else branch is always taken which doubles `size` until the whole string fits and then the string is copied to memory. Maybe I'm misreading the code, in what condition the `then` branch would be taken?

**Review by dpino — APPROVED** [review]
LGTM, left two comments though.

File: src/lib/ptree/trace.lua (hunk 1 above)
Comment (wingo) [reply src/lib/ptree/trace.lua]: Fixed test.

File: src/lib/yang/util.lua (hunk 2 above)
Comment (wingo) [reply src/lib/yang/util.lua:107]: You are right!  I think I was intending to add a method to consume bytes from the front.  That would still be a nice optimization in some cases, so let's maybe keep the code like it is.

**Comment by wingo** [comment]
Addressed feedback and so I merged.  Regarding #1007 I have another fix pending, will post soon.

**Pull request merged**